### PR TITLE
Add script for tag monitor

### PR DESCRIPTION
**Pull request by changev**
This script is for tag monitoring.
It based on a tag-commit history_file and output a property file if there's any changes of tags. 

* When push a tag to a commit, if tag:commit pair never appears before,  tag monitor will catch this change and output a property file.
* There'll be no output when finding deleting tag, only history_file will be updated.
* Tag monitor runs periodically, so if do deleting a tag and pushing a same tag to the commit in one interval, tag monitor won't find any change.

@panpan0000 @PengTian0

### CODE REVIEW

File: manifest-build-tools/application/tag_change_monitor.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    on-tools/manifest-build-tools/application/tag_change_monitor.py \
+    --repo rackhd/on-http \
+    --history_file workspace/on-http_tag_history
+    --property_file workspace/property_file
+    --credential CREDS
+
+The required parameters:
+    repo: user_name/repo_name that indicates which repo should be monitoring.
+    history_file: A file stored one history tag in each line. If doesn't exists it will be created.
+    property_file: A file used for downstream job.
```
Comment (panpan0000) [inline manifest-build-tools/application/tag_change_monitor.py]: as the **output** of script, the comment is not enough...
if I were you , I may say "property_file:  the output of this script, used for downstream Jenkins jobs. this file will contains the newly added tags (comparing with the $hsitory_file), with those tags' corresponding commit/repo_url   "

File: manifest-build-tools/application/tag_change_monitor.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        """
+        with file(self.history_file, 'w') as history_file_handler:
+            for commit, tags in self.new_history.iteritems():
+                tags.insert(0, commit)
+                tags.append('\n')
+                history_file_handler.write(" ".join(tags))
+
+    def write_property_file(self):
+        """
+        write property file for downstream job
```
Comment (panpan0000) [inline manifest-build-tools/application/tag_change_monitor.py]: for those tags "newly" detected for this run,   write those tags' information into a property file ( being used by downstream Jenkins job )

**Comment by PengTian0** [comment]
:+1:

**Comment by changev** [comment]
@anhou @yyscamper 
This PR can be merge now!
Thanks very much!!!

**Pull request merged**